### DISCUSSION TITLE
Autocomplete with lazy option

### DISCUSSION THREAD

**Discussion by omqmo** [taiga-family/taiga-ui#4583] [Q&A] [answered]
Поле ввода, вводишь текст, после перкращения ввода на заданное время выполняется запрос к серверу, а в качестве респонса данные для автокомплита. Вроде все очень просто и даже в доках есть похожие примеры, но я никак не могу  его запустить. Если я использую `tui-input`, то список выподет только после возобнавления ввода. Если использовать `tui-combo-box`, то возникает проблема с невозможностью отключить иконки и пустой список (при фокусе на поле ввода появляется контейнер списка с нулевой высотой и тенью). Прошу направить на  верный путь. Уверен, что это пустяк, который не получается только у меня. Ну и для примера...

```
export class AutocompleteComponent{
  control = new FormControl(``);
  matches$ = this.control.valueChanges.pipe(
    debounceTime<any>( 800 ),
    switchMap( () => this.request() ),
    startWith( [] ),
  );


  request = () => timer(500).pipe(
    map(()=> [`a`, `b`, `c`])
  )
}
```

```
<tui-input #input *tuiLet="matches$ | async as matches" [formControl]="control">
  Placeholder
  <input tuiTextfield type="text"/>

  <ng-container *ngIf="matches?.length">
    <tui-data-list *tuiDataList>
      <button *ngFor="let address of matches" tuiOption [value]="address">
        {{ address }}
      </button>
    </tui-data-list>
  </ng-container>

</tui-input>
```

**Comment by waterplea** [comment]
Сделай, пожалуйста, StackBlitz с тем, что у тебя уже есть, чтобы можно было его поправить.

**Reply by omqmo** [reply]
Вот [репозиторий](https://github.com/omqmo/autocomplete) сейчас еще попытаюсь выложить..

**Reply by omqmo** [reply]
Вроде получилось - https://stackblitz.com/edit/stackblitz-starters-2b5ngj?file=angular.json

**Comment by omqmo** [comment, marked as answer]
Долго до меня доходило, но решение заключается в использовании непосредственно tui-primitive-textfield + tui-hosted-dropdown.
